chameleon/src/main/java/org/hibernate/omm/jdbc/MongoStatement.java

```diff
@@ -136,54 +116,33 @@ public void clearWarnings() throws SimulatedSQLException {
 
     @Override
     @Nullable
-    public ResultSet getResultSet() {
-        return currentQueryResult == null ? null : currentQueryResult.resultSet;
+    public ResultSet getResultSet() throws SimulatedSQLException {
+        throwExceptionIfClosed();
+        throw new NotSupportedSQLException();
     }
 
     @Override
     public int getUpdateCount() {
-        return currentUpdateCount;
+        // unclear what this is, throw for now.
+        throw new UnsupportedOperationException();
     }
 
     @Override
     public boolean getMoreResults() throws SimulatedSQLException {
-        if (currentQueryResult != null) {
```

Comment:
Also unused. But I don't think it has anything to do with MongoDB getMore, which needs to be utilized for any query (this is now taken care of automatically by use of the aggregate helper on MongoCollection that was introduced in MongoStatement

Reply:
agree. I added this initially to fully implement the jdbc semantic but later on I found it is over-engineering.